src/cast_core/src/CollectionScanner.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/CollectionScanner.hpp>
+
+#include <memory>
+#include <chrono>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/MongoException.hpp>
+#include <gennylib/context.hpp>
+
+#include <value_generators/DocumentGenerator.hpp>
+
+
+namespace genny::actor {
+
+struct CollectionScanner::PhaseConfig {
+    mongocxx::database database;
+    std::vector<std::string> collectionNames;
+    bool skipFirstLoop = false;
+    metrics::Operation scanOperation;
```

Comment:
Actually do prefer keeping Operation in the name, because if you have line like:
config->readOperation.start();
vs
config->read.start();

Its a little clearer with operation in the name what is actually happening.